### PR TITLE
Transform Scala example to use a Dockerfile

### LINKED ISSUE #2927: helloworld-scala doesn't use a Dockerfile
## Expected Behavior

Uniformity with other samples, so Dockerfile.

## Actual Behavior

The image is assembled with an sbt plugin.

## Steps to Reproduce the Problem

https://github.com/knative/docs/blob/master/docs/serving/samples/hello-world/helloworld-scala/README.md

### PR DESCRIPTION
**Pull request by markusthoemmes**
Fixes #2927

There's a few more things askew with the Scala sample but I'd like to line those up in a separate PR to keep concerns separate.

### CODE REVIEW

File: docs/serving/samples/hello-world/helloworld-scala/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Use an SBT image matching the Scala and JDK version.
+FROM hseeberger/scala-sbt:8u265_1.4.2_2.12.12 as builder
```
Comment (mattmoor) [inline docs/serving/samples/hello-world/helloworld-scala/Dockerfile:2]: Do you know if there is a multi-arch image we can use here?  All of our other helloworld samples work against arm64 now 😅

Comment (markusthoemmes) [reply docs/serving/samples/hello-world/helloworld-scala/Dockerfile:2]: :thinking: does the build-time image matter, seeing as this is on the JVM?

Comment (mattmoor) [reply docs/serving/samples/hello-world/helloworld-scala/Dockerfile:2]: If I run the Dockerfile build on arm64 it does 😉

Comment (markusthoemmes) [reply docs/serving/samples/hello-world/helloworld-scala/Dockerfile:2]: He, true, quick research didn't yield anything though :/

**Pull request merged**